docs/TESTING.md

```diff
@@ -0,0 +1,208 @@
+---
+title: Testing on particleos
+category: Contributing
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# Testing on particleos
+
+Welcome to the **particleOS Testing and Validation Guide**. This document
+outlines the comprehensive procedure for developers and contributors, covering
+the entire workflow from building the particle OS image to performing an
+installation on a simulated empty drive and finally testing the resulting
+installed system within a Virtual Machine environment.
+
+The goal is to provide clear, actionable instructions to facilitate a smooth
+setup of your development environment, enabling effective building,
+installation, and testing of changes to the particleOS project.
+
+Welcome to the particleos testing guide! This document provides the step by step
+guide from the particleOS image creation to the test in Virtual Machine
+of the final installed image, going through the installation process on an
+empty drive.
+
+It aims to be a comprehensive guide for developers who want to contribute to the
+project by providing clear instructions on how to set up a development
+environment, build the project, and test their changes.
+
+## Prerequisites
+
+### Environment setup
+
+* Before you start testing on particleos, make sure to have the **lastest**
+version of `mkosi` accessible from your path-environment.
+
+  ```bash
+  cd ${WORKDIR}
+  git clone https://github.com/systemd/mkosi.git
+  export PATH="${WORKDIR}/mkosi/bin/:${PATH}"
+  ```
+
+* `mkosi` requires Python 3.8 or higher. You can install the lastest python
+version and force the Interpreter as follow:
+
+  ```bash
+  export MKOSI_INTERPRETER="/usr/bin/python3.12"
+  ```
+
+* Verify the version of mkosi installed:
+
+  ```bash
+  mkosi --version
+  ```
+
+* You will also need `qemu` installed on your system to test the created images in
+a virtual machine.
+
+### Lastest systemd version (optional but recommended)
+
+As documented in the main `README.md`, it is highly recommended to use the
+latest version of `systemd`. Choose one of the following methods:
+
+* [Using the OBS profile](../README.md#using-the-obs-profile-to-fetch-a-newer-systemd)
+* [Building systemd from source locally](../README.md#building-systemd-from-source)
+
+## Building the particleOS Image
+
+To build the particleOS Image:
+
+1. Navigate to the root of the particleos repository and create the local
+  config `mkosi.local.conf` file with the
+  following content:
+
+  ```conf
+  [Match]
+  Distribution=opensuse
+
+  [Distribution]
+  Release=tumbleweed
+
+  [Runtime]
+  VirtualMachineMonitor=qemu
+
+  QemuArgs=
+          -drive if=none,file=./mkosi.output/target-disk.img,format=raw,id=installdisk
+          -device virtio-blk-pci,drive=installdisk
+  ```
```

Comment:
This flat out doesn't work on any distribution other than opensuse because of the `[Match]`